mamba_ssm/ops/selective_scan_interface.py

```diff
@@ -301,11 +305,11 @@ def mamba_inner_fn(
     xz, conv1d_weight, conv1d_bias, x_proj_weight, delta_proj_weight,
     out_proj_weight, out_proj_bias,
     A, B=None, C=None, D=None, delta_bias=None, B_proj_bias=None,
-    C_proj_bias=None, delta_softplus=True
+    C_proj_bias=None, mask=None, delta_softplus=True
 ):
     return MambaInnerFn.apply(xz, conv1d_weight, conv1d_bias, x_proj_weight, delta_proj_weight,
```

Comment:
I have some confusion about this line of code. Since MambaInnerFn doesn't provides parameters option for `mask`, it seems that it has no effect one the fwd and bwd pass.
Hence, mask doesn't seem to work.

Reply:
I might be wrong, but the PR modified Line 162 in the very same file, which is the definition of `MambaInnerFn.forward` method. Also, multiplications on Line 181 and 220 used `mask`.

Reply:
Thank you! I missed this line of code.
But I see in this PR, `attention_mask` is used in the forward pass, but seem not to be used in the backward pass. So when I tried to feed in batch data with left padding and masking (batch_size, seq_len, hidden_dim) into mamba block , it reports error. Has anyone encountered a similar error?

Err Msg:
```
  File "/blahblah/miniconda3/envs/dev/lib/python3.10/site-packages/torch/_tensor.py", line 488, in backward
    torch.autograd.backward(
  File "/blahblah/miniconda3/envs/dev/lib/python3.10/site-packages/torch/autograd/__init__.py", line 197, in backward
    Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: function MambaInnerFnBackward returned an incorrect number of gradients (expected 16, got 15)
```

Reply:
@xtwigs pointed out that one can add `None` in the returned gradients to fix the issue. (For fellows above who wonder where to put `None`, I did it at the end of the tuple.) I agree with @xtwigs as we don't calculate gradient on the mask tensor.

> > > I tried using this branch but got an error about not getting expected number of gradients during backward (15 vs 16)
> > 
> > 
> > Yeah, I got the same error. Does anyone know how to solve it?
> 
> I think this error happens because Pytorch expects a gradient for the mask to be returned, which isn't the case here. Adding a `None` in `mamba_ssm/ops/selective_scan_interface.py#301` fixes this issue.
> 
> On the other hand to bypass the recalculation of conv1d in the backward pass we can set checkpoint_level to 0 (?).
> 
> Can someone verify if my thought process is accurate?